Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -50,6 +50,56 @@ Building a reverse engineering environment for free is possible. However, there
 
 Because Objective-C and Swift are fundamentally different, the programming language in which the app is written affects the possibilities for reverse engineering it. For example, Objective-C allows method invocations to be changed at runtime. This makes hooking into other app functions (a technique heavily used by [Cycript](http://www.cycript.org/ "Cycript") and other reverse engineering tools) easy. This "method swizzling" is not implemented the same way in Swift, and the difference makes the technique harder to execute with Swift than with Objective-C.
 
+In iOS, all the application code, for both Swift and Objective-C, is compiled to native, unlike Android where code is compiled into Dalvik Bytecode and native code (if code written in C/C++). Thus, to analyze iOS application a disassembler is needed. 
+
+Before getting into how to disassemble an application's code, it is important to know that the application downloaded via Apple Appstore is encrypted using Fairplay DRM and need to be decrypted before it can be analyzed. This has been already discussed in "[Acquiring the App Binary](0x06b-basic-security-testing.md#acquiring-the-app-binary "Acquiring the App Binary")" section. In this section we assume you already have access to a decrypted iOS application.
+
+In this section the term "app binary" refers to the Macho-O file in the application bundle which contains the compiled code, and should not be confused with the application bundle - the IPA file. Composition of an application bundle has already been discussed in "[Exploring the App Package](0x06b-basic-security-testing.md#exploring-the-app-package "Exploring the App Package")" section. 
+
+##### Disassembling With Ghidra
+As mentioned in "[Tooling](#tooling "Tooling")" section, Ghidra is an open source software reverse engineering tool. Among multiple functionalities offered by Ghidra, it can also be used to analyze iOS application binaries. 
+
+To get started with Ghidra is easy. Start Ghidra using *ghidraRun* (\*nix) or *ghidraRun.bat* (Windows), depending on the platform you are on. Once Ghidra is fired up, create a new project by specifying the project directory. You will be greeted by a window as shown below:  
```

Comment:
Please have a look at the [style guide](https://github.com/OWASP/owasp-mstg/blob/master/style_guide.md), maybe you can change it to:
```suggestion
To get started with Ghidra is easy. Start Ghidra using `ghidraRun` (\*nix) or `ghidraRun.bat` (Windows), depending on the platform you are on. Once Ghidra is fired up, create a new project by specifying the project directory. You will be greeted by a window as shown below:  
```